test/Microsoft.AspNet.Mvc.Core.Test/Filters/FormatFilterTest.cs

```diff
@@ -0,0 +1,324 @@
+﻿using System;
+using Microsoft.AspNet.Http;
+using Microsoft.AspNet.Mvc;
+using Microsoft.AspNet.PipelineCore;
+using Microsoft.AspNet.Routing;
+using Microsoft.Framework.DependencyInjection;
+using Microsoft.Framework.OptionsModel;
+using Microsoft.Net.Http.Headers;
+using Xunit;
+
+#if ASPNET50
+using Moq;
+#endif
+
+namespace Microsoft.AspNet.Mvc.Test
+{
+    public class FormatFilterTests
+    {
+        public enum FormatSource
+        {
+            RouteData,
+            QueryData,
+            RouteAndQueryData
+        }
+
+        [Theory]
+        [InlineData("json", FormatSource.RouteData, "application/json")]
+        [InlineData("json", FormatSource.QueryData, "application/json")]
+        [InlineData("json", FormatSource.RouteAndQueryData, "application/json")]
```

Comment:
Is this case the same as `FormatFilter_ContextContainsFormat_InRouteAndQueryData`?